### PR TITLE
Changed to be compatible with all screen sizes

### PR DESCRIPTION
**Pull request by Cody-Soultz**
Changed to be compatible with all screen sizes however the right side needs to be in a scroll pane or needs to be scaled based on height of window.

### CODE REVIEW

File: EvolvioColor/EvolvioColor.pde [hunk 1]
```diff
@@ -4,10 +4,10 @@ final float NOISE_STEP_SIZE = 0.1;
 final int BOARD_WIDTH = 100;
 final int BOARD_HEIGHT = 100;
 
-final int WINDOW_WIDTH = 1920;
-final int WINDOW_HEIGHT = 1080;
+int WINDOW_WIDTH;
```
Comment (Cody-Soultz) [inline EvolvioColor/EvolvioColor.pde]: I had to get rid of the final tags so that I could assign the value later when we could use the variables displayWidth and displayHeight

File: EvolvioColor/EvolvioColor.pde [hunk 2]
```diff
@@ -78,6 +82,7 @@ void draw() {
     evoBoard.fileSave();
     prevMouseX = mouseX;
     prevMouseY = mouseY;
+    surface.setSize(displayWidth,displayHeight);
```
Comment (Cody-Soultz) [inline EvolvioColor/EvolvioColor.pde]: This is not needed, however it is useful because the UI does not respond well to size changes so this makes the size never change, but the window itself can be docked and undocked at least in windows 10.

Comment (CrazyPython) [inline EvolvioColor/EvolvioColor.pde]: @Cody-Soultz use line comments instead of reviews

**Comment by CrazyPython** [comment]
@Cody-Soultz  make first letter of comment lowercase and space between `//` and the comment. Better code style. And put the comment on top of all the lines instead of duplicating them for each line.

why'd you close?

**Comment by Cody-Soultz** [comment]
Incorrectly closed this PR
I am reopening #9 and closing #15 since they are duplicates

**Comment by Cody-Soultz** [comment]
@CrazyPython  What do you mean by `` And put the comment on top of all the lines instead of duplicating them for each line.

**Comment by CrazyPython** [comment]
```
codecodecode //commentcommentcomment
morecodecodecode //samecommentsamecommentsamecomment
yetmorecodecodecode //the same comment again
```
->

```
// commentcommentcomment
codecodecode
morecodecodecode
yetmorecodecodecode
```
space between comment & `//`

**Comment by YoYoYonnY** [comment]
@CrazyPython You are severely over-complicating this. @Cody-Soultz's use of PR review is correct and IMHO even preferred.

**Comment by YoYoYonnY** [comment]
I was about to merge, when I noticed that the changes do not actually work as intended. If anyone (preferably on Windows) can confirm that these changes **do** work, I will reopen and pull, but for now I will postpone. Closing.

**Comment by Cody-Soultz** [comment]
Can you tell me what the issue is? I had no problem on my windows based machine.

**Comment by Cody-Soultz** [comment]
I just pushed a version of my code including all the other PR that you have added. I think what you meant by the changes not actually working as intended you were referring to how the code didn't truly dock in windows as expected. I found that bug and fixed it by removing surface.setSize(displayWidth,displayHeight); from the bottom of the draw function. Please reopen this PR

**Comment by Cody-Soultz** [comment]
Let me know if there is still an issue. I just hated the hard coded screen size. I know that creates other problems to make the UI look good at any size, however for my laptop's screen the set size was too big and half of the view was off the screen.

**Comment by YoYoYonnY** [comment]
@Cody-Soultz The problem I was having is that resizing moved the window but did not actually change it's dimensions. I will take a look at it again.

**Comment by Cody-Soultz** [comment]
Ok yeah that is what that one line of code did. It stopped the user from resizing since the UI has not been designed for resizing yet.

**Comment by YoYoYonnY** [comment]
@Cody-Soultz Due to the changes suggested by @CrazyPython it's become a pain to merge your fork. I will try to pull your changes as soon as possible.

**Comment by Cody-Soultz** [comment]
And as I said earlier the right side with the buttons and graph needs to be scaled down or have a scroll pane because on smaller screens the bottom of the right side gets cut off even before my change.
capture

**Comment by YoYoYonnY** [comment]
@Cody-Soultz True... Not sure if I should put your changes in a separate branch, or merge to the master branch and report the issue. I'll see what @mbertheau has to say.

**Comment by YoYoYonnY** [comment]
IMHO https://github.com/Cody-Soultz/evolv.io/commit/bcc8be265ef0a3179340ffbd84a77a73ac310327 and https://github.com/Cody-Soultz/evolv.io/commit/148652a888f1df196e789205e79f48ad57eda6cc add no value to the project, while cluttering the code and the commit history. Furthermore, https://github.com/Cody-Soultz/evolv.io/commit/f526a478302e3cea9aad32ee4cbd7992c676e129 makes it harder to pull your latest commit, https://github.com/Cody-Soultz/evolv.io/commit/9930b896883349009367454083a890a14611f10d, without that one.

I suggest squashing and force pushing only your first and last commit, so the pull request succeeds without me having to revert the 'dirty' commits, or, god forbid, having to rewrite commit history. If you give me a second though, if you gave me writing permission to your fork, I might be able to do that for you.

**Comment by Cody-Soultz** [comment]
Ok Either that or I can use the cli to do that, but I don't have much experience for squashing. I do have experience rolling back and changing commit history though (I only use CLI I hate git GUI's). I gave you write access I think

**Comment by Cody-Soultz** [comment]
I am just new to PR on open source projects like this. I have only done these internally. In the future I will make a new branch so that I don't have to wait in limbo while the PR is accepted or rejected. I know I could make a new branch now but then I have to roll back if this change is not accepted.

**Comment by Cody-Soultz** [comment]
Also, while it is being talked about the graph on the left side resizes to a smaller screen perfectly, unless the GUI gets squeezed in horizontally, but I think there is an initial mechanism to make it fit perfectly to a certain percent of the screen so if there is a way to fire that mechanism on screen resize then the left side would scale perfectly.

**Comment by Cody-Soultz** [comment]
Ok I just cleaned it up by reverting to original fix adding comments and getting rid of the buggy line of code. Then I committed and did a $ git push -f to erase the history of merging in the most recent version of the evolv.io code

**Comment by YoYoYonnY** [comment]
@Cody-Soultz Could you remove the comment on line 7 and do that again? If you do that, I will be able to merge the pull request. I'm still unsure if I should merge to master, though.

**Comment by YoYoYonnY** [comment]
Sorry, spoke too soon... wait...

**Comment by YoYoYonnY** [comment]
Could you please remove the comment at line 7 of EvolvioColor/EvolvioColor.pde and do that again? After that I will merge. I don't like merging bugged changes to the master branch, but it's the best solution I've got.

**Comment by Cody-Soultz** [comment]
OK

**Comment by Cody-Soultz** [comment]
Done. Is that everything for this PR?

**Comment by YoYoYonnY** [comment]
Yes, the style is messed up again, but I can change that.

**Comment by YoYoYonnY** [comment]
Changing style is not quite as bad as reverting changes just merged.

**Comment by Cody-Soultz** [comment]
Oh yeah I see the style issue, sorry on my notepad++ it does not show that. I also squashed it all to one commit.

**Comment by YoYoYonnY** [comment]
But next time please send your pull request _after_ you've tested it works. :unamused:

**Comment by YoYoYonnY** [comment]
OK now your code is bugged again, wait a second, let me fix this.

**Comment by YoYoYonnY** [comment]
Also, we should probably move this conversation to a chat room, but considering no such thing exists, lets keep talking like this for now

**Comment by Cody-Soultz** [comment]
OK, I think it was more an issue of what we call right I guess....I had thought it was better to not let the user change the size of the screen rather than to let them change the size and the UI be messed up. What went wrong this time? It works on my end.

**Comment by YoYoYonnY** [comment]
Oh, I see, I was looking at the wrong diff

**Comment by YoYoYonnY** [comment]
And the style was messed up because you used tabs instead of spaces

**Comment by Cody-Soultz** [comment]
I did create a bug to get the UI fixed

**Comment by Cody-Soultz** [comment]
Oh I see lol I prefer tabs to spaces any day, but I guess to each there own you know

**Comment by YoYoYonnY** [comment]
You seem to be pushing the changes faster than me. Can you turn the tabs into spaces?

**Comment by Cody-Soultz** [comment]
Yeah...should I squash it again. and what lines in specific so that I can quickly do it

**Comment by YoYoYonnY** [comment]
Let me do it then...

**Comment by YoYoYonnY** [comment]
Please don't force push for a second...

**Comment by Cody-Soultz** [comment]
OK...I just saw them with a git diff lol

**Comment by YoYoYonnY** [comment]
Let's hope for the best..

**Comment by YoYoYonnY** [comment]
OK, looks good. Finally, merging. Let's hope that the pull request still closes...

**Comment by YoYoYonnY** [comment]
Finally!

**Comment by Cody-Soultz** [comment]
Can you then label the issue I created?

**Comment by YoYoYonnY** [comment]
Done.

**Comment by Cody-Soultz** [comment]
Thank You

**Comment by YoYoYonnY** [comment]
@Cody-Soultz: For the future: Please only send a pull request when the feature is implemented. Next time you don't have to expect me or whoever is handling your pull request to keep your request open until your changes are finished. Forget everything @CrazyPython has said here and in #15: Pull requests do _not_ update as your fork does, a pull request is a request to merge a _range of commits_, not a _fork_. If you do intend to merge your entire fork, you should close the request and create a new one, or ask and hope that someone merges the rest of the fork for you. Also, please use PR review and commit messages instead of source comments whenever possible.

**Comment by Cody-Soultz** [comment]
OK will do thank you

**Comment by mbertheau** [comment]
@Cody-Soultz and @YoYoYonnY Thank you for your work on this. I think the changes in this PR remove a potential obstacle for new users that might become contributors, so that's a good thing.

However, cc @CrazyPython, let me differ on how I understand @YoYoYonnY 's stance on PRs.

1. It's perfectly ok to send a PR when it's not ready, to get early commentary and/or testing input from other people. Just make it clear in the description what you're looking for (a merge, comments, testing, ...)
2. PRs are a a request to merge a branch ("feature branch"), that is usually in your fork of this repository, to a branch (usually master) in this repository. If you add commits to the feature branch, rewrite its history or rebase it when the master branch of this repo was updated, the PR will reflect these changes accordingly. Therefore adding commits to the feature branch is perfectly ok and these added commits will be merged automatically when the PR is merged.
3. There's a place for every comment, sometimes it's a GitHub review comment, a commit message, or a source code comment.

Transient comments that lose their value once the issue they describe is resolved, belong in a GitHub review comment, if they should be resolved before the PR is merged.

Commit messages should on the first line mention concisely what was changed, then an empty line, and then explain why the change was made. Often the explanation is as simple as "that was obviously a bug and fixed it", sometimes it's just "I added this feature that we think makes sense", in which case there's not much of an explanation to write. However sometimes it's not that obvious, and that is when a more detailed explanation makes sense.

If it's not trivially obvious how the code change achieves what it achieves, then that should be explained as well.

Some people say that the first line of the commit message should start with a third person singular verb (i.e. "Add foo", not "Adds foo" or "Added foo"), and I agree with them, but I'm not religious about it, since this is a hobby project that we all want to have some fun with :)

Finally source code comments. In my opinion code that doesn't need comments to be understood is better than code that does. However sometimes the issues at hand are tricky enough, or it's a long mental way from the means (code change) to the end (what is achieved with the code change). In that case a source code comment helps the reader understand the code.

In conclusion: Open PRs as soon as you have something, and let us know your intentions. Maybe you don't know git or github well and you don't have the time to learn it now, then we're happy to pull your changes and make necessary adjustments ourselves. Maybe you want to not only contribute something to evolv.io but also learn something about contributing on GitHub in general. We'll be happy to provide assistance with that as well :)

**Comment by CrazyPython** [comment]
@YoYoYonnY Do you have any evidence that a pull request will not automatically include new commits?

**Comment by YoYoYonnY** [comment]
@CrazyPython As I explained earlier, a pull request will succeed if its initial commits (Or just the first commit, though I have never merged a pull request with removed commits) have been pushed.

Furthermore, A pull request is not a mechanism, it's a concept: You send a pull request when you want your changes (commits) to be pulled (merged) into the repository. GitHub's pull requests are just a medium to do so. Some projects use email, web forums, external programs (e.g. GitHub Desktop), etc.

Therefore, adding commits to your fork doesn't guarantee they will be merged into the repository, and it definitely doesn't mean they will be added to the pull request, which, after all, is the concept of requesting changes to be pulled.

For example, I personally would like to fetch your fork, review the changes you made, then push or close the request, whichever is applicable.

If you in the meantime added commits to your fork, I might close your request because it is incomplete, or, for big pull requests, because it does not merge, while you, in the meantime, have pushed commits to your fork which fixed those problems.

Worse, I might push your changes, not _knowing_ your pull request is unfinished.
I might then merge something which will completely break everything but still pass the review/unit tests, even though your fork is clean.

Worse yet, I might chose to use the merge button on GitHub to merge the changes (Which, by the way, is a bad idea for any important repository. The GitHub merge button is meant to make merging easier and faster, not safer or cleaner).
If you added malicious commits after my review is completed, I will unknowingly merge them with the rest.

Of course, all of this can be avoided, but even a foolproof system will slow merging pull request down, and, by extension, the whole project.

**Comment by CrazyPython** [comment]
@YoYoYonnY The thing is, I have experimental evidence that adding commits on top of my fork does indeed update the pull request.
